content/doc/developer/testing/index.adoc

```diff
@@ -110,7 +110,29 @@ Most Java IDEs should be able to run JUnit tests and report on the results.
 == What to Test
 Now that we can write a basic test, we should discuss what you should be testing…
 
-TODO: Unit testing of your code, as much as possible. JenkinsRule testing: creating jobs that use your build steps, running them, asserting on the output
+The test pyramid is a concept used in software development to describe the ideal distribution of different types of automated tests that make up a test suite.
+The idea was introduced by Mike Cohn and is visualized as a pyramid to help illustrate the optimal proportion of each type of test.
+Here's a breakdown of the three main layers from the bottom to the top of the pyramid:
+
+1. *Unit Tests*: At the base of the pyramid, unit tests are the most numerous.
+These tests are focused on checking the smallest parts of an application, such as individual functions or methods.
+Unit tests are quick to execute and help ensure that each component of the software works as expected in isolation.
+
+2. *Integration Tests*: The middle layer of the pyramid consists of integration tests.
+These are fewer in number compared to unit tests.
+Integration tests verify that different parts of the system work together as intended.
+For Jenkins, this makes use of the link:https://javadoc.jenkins.io/component/jenkins-test-harness/org/jvnet/hudson/test/JenkinsRule.html[`JenkinsRule`] to test the integration of the plugin into Jenkins.
+This can be a test for a new build step and testing this with a test pipeline.
+
+3. *End-to-End Tests (E2E) and UI Tests*: At the top of the pyramid, end-to-end tests are the fewest in number but typically the most complex.
+These tests simulate real user scenarios from start to finish.
+They interact with the application as a user would, testing the complete flow of the system.
+For Jenkins, this makes use of the link:https://javadoc.jenkins.io/component/jenkins-test-harness/org/jvnet/hudson/test/JenkinsRule.html[`JenkinsRule`] or link:https://javadoc.jenkins.io/component/jenkins-test-harness/org/jvnet/hudson/test/JenkinsSessionRule.html[`JenkinsSessionRule`] in combination with the link:https://javadoc.jenkins.io/component/jenkins-test-harness/org/jvnet/hudson/test/JenkinsRule.html#createWebClient()[`createWebClient()`]-call which is used to really creates a call to endpoint to assert something on the result.
```

Comment:
```suggestion
For Jenkins, this makes use of the link:https://javadoc.jenkins.io/component/jenkins-test-harness/org/jvnet/hudson/test/JenkinsRule.html[`JenkinsRule`] or link:https://javadoc.jenkins.io/component/jenkins-test-harness/org/jvnet/hudson/test/JenkinsSessionRule.html[`JenkinsSessionRule`] in combination with the link:https://javadoc.jenkins.io/component/jenkins-test-harness/org/jvnet/hudson/test/JenkinsRule.html#createWebClient()[`createWebClient()`]-call, which is used to create a call to an endpoint to then assert something on the result.
```
Small suggestion to adjust the wording of the last line here, as its current syntax is not clear.